tests/unit/test_rest_streaming.py

```diff
@@ -114,103 +120,201 @@ def iter_content(self, *args, **kwargs):
-@pytest.mark.parametrize("random_split", [False])
-def test_next_simple(random_split):
-    responses = [EchoResponse(content="hello world"), EchoResponse(content="yes")]
+@pytest.mark.parametrize(
+    "random_split,resp_message_is_proto_plus,response_type",
+    [(False, True, EchoResponse), (False, False, httpbody_pb2.HttpBody)],
+)
+def test_next_simple(random_split, resp_message_is_proto_plus, response_type):
+    if resp_message_is_proto_plus:
+        responses = [EchoResponse(content="hello world"), EchoResponse(content="yes")]
+    else:
+        responses = [
+            httpbody_pb2.HttpBody(content_type="hello world"),
+            httpbody_pb2.HttpBody(content_type="yes"),
+        ]
+
     resp = ResponseMock(
-        responses=responses, random_split=random_split, response_cls=EchoResponse
+        responses=responses, random_split=random_split, response_cls=response_type
     )
-    itr = rest_streaming.ResponseIterator(resp, EchoResponse)
+    itr = rest_streaming.ResponseIterator(resp, response_type)
     assert list(itr) == responses
 
 
-@pytest.mark.parametrize("random_split", [True, False])
-def test_next_nested(random_split):
-    responses = [
-        Song(title="some song", composer=Composer(given_name="some name")),
-        Song(title="another song", date_added=datetime.datetime(2021, 12, 17)),
-    ]
+@pytest.mark.parametrize(
+    "random_split,resp_message_is_proto_plus,response_type",
+    [
+        (True, True, Song),
+        (False, True, Song),
+        (True, False, http_pb2.HttpRule),
+        (False, False, http_pb2.HttpRule),
+    ],
+)
+def test_next_nested(random_split, resp_message_is_proto_plus, response_type):
+    if resp_message_is_proto_plus:
+        responses = [
+            Song(title="some song", composer=Composer(given_name="some name")),
+            Song(title="another song", date_added=datetime.datetime(2021, 12, 17)),
+        ]
+    else:
+        # Although `http_pb2.HttpRule`` is used in the response, any response message
+        # can be used which meets this criteria for the test of having a nested field.
+        responses = [
+            http_pb2.HttpRule(
+                selector="some selector",
+                custom=http_pb2.CustomHttpPattern(kind="some kind"),
+            ),
+            http_pb2.HttpRule(
+                selector="another selector",
+                custom=http_pb2.CustomHttpPattern(path="some path"),
+            ),
+        ]
     resp = ResponseMock(
-        responses=responses, random_split=random_split, response_cls=Song
+        responses=responses, random_split=random_split, response_cls=response_type
     )
-    itr = rest_streaming.ResponseIterator(resp, Song)
+    itr = rest_streaming.ResponseIterator(resp, response_type)
     assert list(itr) == responses
 
 
-@pytest.mark.parametrize("random_split", [True, False])
-def test_next_stress(random_split):
+@pytest.mark.parametrize(
+    "random_split,resp_message_is_proto_plus,response_type",
```

Comment:
Ditto on parametrization

Reply:
Fixed in https://github.com/googleapis/python-api-core/pull/604/commits/5b7a4ce57b7df5dab1979dbe8151fe2e726dc7e5